providers/aws/resources/aws_ec2.go

```diff
@@ -400,104 +408,25 @@ func (a *mqlAwsEc2) getSecurityGroups(conn *connection.AwsConnection) []*jobpool
-							map[string]*llx.RawData{
-								"id":         llx.StringData(convert.ToString(group.GroupId) + "-" + strconv.Itoa(p) + "-egress"),
-								"fromPort":   llx.IntDataDefault(permission.FromPort, -1),
-								"toPort":     llx.IntDataDefault(permission.ToPort, -1),
-								"ipProtocol": llx.StringDataPtr(permission.IpProtocol),
-								"ipRanges":   llx.ArrayData(ipRanges, types.Any),
-								"ipv6Ranges": llx.ArrayData(ipv6Ranges, types.Any),
-							})
-						if err != nil {
-							return nil, err
-						}
-
-						mqlIpPermissionsEgress = append(mqlIpPermissionsEgress, mqlSecurityGroupIpPermission)
-					}
-
 					args := map[string]*llx.RawData{
-						"arn":                 llx.StringData(fmt.Sprintf(securityGroupArnPattern, regionVal, conn.AccountId(), convert.ToString(group.GroupId))),
-						"id":                  llx.StringDataPtr(group.GroupId),
-						"name":                llx.StringDataPtr(group.GroupName),
-						"description":         llx.StringDataPtr(group.Description),
-						"tags":                llx.MapData(Ec2TagsToMap(group.Tags), types.String),
-						"ipPermissions":       llx.ArrayData(mqlIpPermissions, types.Resource("aws.ec2.securitygroup.ippermission")),
-						"ipPermissionsEgress": llx.ArrayData(mqlIpPermissionsEgress, types.Resource("aws.ec2.securitygroup.ippermission")),
-						"region":              llx.StringData(regionVal),
+						"arn":         llx.StringData(fmt.Sprintf(securityGroupArnPattern, regionVal, conn.AccountId(), convert.ToString(group.GroupId))),
+						"id":          llx.StringDataPtr(group.GroupId),
+						"name":        llx.StringDataPtr(group.GroupName),
+						"description": llx.StringDataPtr(group.Description),
+						"tags":        llx.MapData(Ec2TagsToMap(group.Tags), types.String),
+						"region":      llx.StringData(regionVal),
 					}
 
-					mqlVpc, err := NewResource(a.MqlRuntime, "aws.vpc",
-						map[string]*llx.RawData{
-							"arn": llx.StringData(fmt.Sprintf(vpcArnPattern, regionVal, conn.AccountId(), convert.ToString(group.VpcId))),
-						})
+					mqlEc2SecurityGroup, err := CreateResource(a.MqlRuntime, "aws.ec2.securitygroup", args)
 					if err != nil {
 						return nil, err
 					}
-					if mqlVpc != nil {
-						args["vpc"] = llx.ResourceData(mqlVpc, mqlVpc.MqlName())
-					} else {
-						args["vpc"] = llx.NilData
-					}
-					mqlS3SecurityGroup, err := CreateResource(a.MqlRuntime, "aws.ec2.securitygroup", args)
-					if err != nil {
-						return nil, err
-					}
-					res = append(res, mqlS3SecurityGroup)
+					res = append(res, mqlEc2SecurityGroup)
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).cacheIpPerms = group.IpPermissions
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).cacheIpPermsEgress = group.IpPermissionsEgress
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).groupId = *group.GroupId
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).region = region
+					mqlEc2SecurityGroup.(*mqlAwsEc2Securitygroup).cacheVpc = group.VpcId
```

Comment:
does that persist across query runs? this updates the MQL object after it's been stored

Reply:
ya, I believe so. we use this same pattern a lot in the aws code